src/Microsoft.Diagnostics.Monitoring.EventPipe/Counters/TraceEventExtensions.cs

```diff
@@ -23,6 +23,39 @@ public static bool TryGetCounterPayload(this TraceEvent traceEvent, CounterFilte
                 string series = payloadFields["Series"].ToString();
                 string counterName = payloadFields["Name"].ToString();
 
+                var metadataDict = new Dictionary<string, string>();
+
+                ReadOnlySpan<char> metadata = payloadFields["Metadata"].ToString();
+
+                while (!metadata.IsEmpty)
```

Comment:
If this custom implementation is going to be used (I'm fine with using `string.Split`, it's only one allocation of `string[]` and the rest of the internals is already using `ReadOnlySpan`), then I think (1) this should be made into a separate static method and (2) add unit testing to make sure the custom algorithm works as intended.